homeassistant/components/emoncms/config_flow.py

```diff
@@ -0,0 +1,170 @@
+                errors["base"] = result[CONF_MESSAGE]
+            else:
+                title = sensor_name(user_input[CONF_URL])
+                return self.async_create_entry(title=title, data=user_input)
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_URL): str,
+                        vol.Required(CONF_API_KEY): str,
+                    }
+                ),
+                user_input,
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_info: ConfigType) -> ConfigFlowResult:
+        """Import config from yaml."""
+        url = import_info[CONF_URL]
+        api_key = import_info[CONF_API_KEY]
+        include_only_feeds = None
+        if import_info.get(CONF_ONLY_INCLUDE_FEEDID) is not None:
+            include_only_feeds = list(map(str, import_info[CONF_ONLY_INCLUDE_FEEDID]))
+        if not include_only_feeds:
+            emoncms_result = await get_feed_list(self.hass, url, api_key)
+            if emoncms_result[CONF_SUCCESS]:
+                include_only_feeds = [
+                    feed[FEED_ID] for feed in emoncms_result[CONF_MESSAGE]
+                ]
+        config = {
+            CONF_API_KEY: api_key,
+            CONF_ONLY_INCLUDE_FEEDID: include_only_feeds,
+            CONF_URL: url,
+        }
+        LOGGER.debug(config)
+        result = await self.async_step_user(config)
+        if errors := result.get("errors"):
+            return self.async_abort(reason=errors["base"])
+        return result
+
+
+class EmoncmsOptionsFlow(OptionsFlowWithConfigEntry):
+    """Emoncms Options flow handler."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options."""
+        errors: dict[str, str] = {}
+        url = self._config_entry.data[CONF_URL]
+        api_key = self._config_entry.data[CONF_API_KEY]
+        include_only_feeds = self._config_entry.data.get(CONF_ONLY_INCLUDE_FEEDID, [])
+        options: Any = include_only_feeds
+        result = await get_feed_list(self.hass, url, api_key)
+        if not result[CONF_SUCCESS]:
+            errors["base"] = result[CONF_MESSAGE]
+        else:
+            options = get_options(result[CONF_MESSAGE])
+        dropdown = {"options": options, "mode": "dropdown", "multiple": True}
+        if user_input:
+            url = user_input[CONF_URL]
+            api_key = user_input[CONF_API_KEY]
+            include_only_feeds = user_input[CONF_ONLY_INCLUDE_FEEDID]
+            if self.hass.config_entries.async_update_entry(
+                self._config_entry, title=sensor_name(url), data=user_input
+            ):
+                LOGGER.debug("entry updated")
+            return self.async_create_entry(data=user_input)
```

Comment:
We should store this in the options instead, currently if you would submit the options flow it would reload the integration twice. So please only use the `self.async_create_entry`

Reply:
done